### DISCUSSION TITLE
GraphicsView with panning

### DISCUSSION THREAD

**Discussion by kj4bix** [dotnet/maui#15403] [Q&A]
I have an application I'm porting from Xamarin.Forms to MAUI. Previously I had my own device specific graphics routines that did drawing and transformation of vectors into a Canvas. The GraphicsView is great because I don't need to do that any more, however my previous app also had zooming and panning of the objects. When I attempted the pan gestures for the GraphicsView it no longer clips the graphics to the drawing area as defined. Is there a way to force clipping of the graphics to the graphicsview window? 

Here is a display of what happens on Windows, the graphics just move outside the defined window - not what I would normally expect. 
<img width="1716" alt="pan issue" src="https://github.com/dotnet/maui/assets/9199409/6bc94f9a-af3d-4977-85fa-fcda37a9da82">

**Comment by xtuzy** [comment]
try IsClippedToBounds, canvas also have a clip method

**Reply by kj4bix** [reply]
No change on Windows. I have the GraphicsView inside a ContentView - I added the IsClippedToBounds on the ContentView and it had no effect. I don't notice a IsClippedToBounds on the GraphicsView itself. Attempting a Canvas.Clip crashes on Windows using the size of the rectangle.

**Reply by xtuzy** [reply]
Can you provide examples? i try use translate method, don't get error
